src/ansys/turbogrid/core/multi_blade_row/multi_blade_row.py

```diff
@@ -0,0 +1,1130 @@
+    write_tginit_first = False
+
+    #: When working in Ansys Labs, the number of retry to be made for file transfer from
+    #: container to the local folder.
+    max_file_transfer_attempts = 20
+
+    #: When working in Ansys Labs, the name of the file containing the container key with full path.
+    tg_container_key_file = ""
+
+    def __init__(self, working_dir: str, case_directory: str, ndf_file_name: str):
+        """
+        Initialize the class using name with full path of an NDF file for the multi blade row case.
+
+        Parameters
+        ----------
+        ndf_file_full_name : str
+            Name with full path of the NDF file containing the blade rows for the multi blade row case.
+        """
+        self._set_working_directory(working_dir)
+        self._set_case_directory_and_ndf(case_directory, ndf_file_name)
+        self._ndf_parser = ndfp.NDFParser(self._ndf_file_full_path)
+        self._assert_unique_blade_names()
+        self.set_blade_rows_to_mesh([])
+
+    def set_blade_rows_to_mesh(self, blades_to_mesh: list):
+        """
+        Set the blade rows to be meshed.
+
+        Parameters
+        ----------
+        blades_to_mesh:list
+            Names of the main blade from each blade row to be meshed in a list.
+        """
+        all_blade_rows = self._ndf_parser.get_blade_row_blades()
+        blade_row_names_to_mesh = [
+            x
+            for x in all_blade_rows
+            if (len(blades_to_mesh) == 0 or all_blade_rows[x][0] in blades_to_mesh)
+        ]
+        self._blade_rows_to_mesh = {}
+        for blade_row in blade_row_names_to_mesh:
+            self._blade_rows_to_mesh[blade_row] = all_blade_rows[blade_row]
+        # print(f"Blade rows to mesh: {self._blade_rows_to_mesh.keys()}")
+        # for blade_row in self._blade_rows_to_mesh:
+        #    print(f"Blade row {blade_row} blades: {self._blade_rows_to_mesh[blade_row]}")
+
+    def set_multi_process_count(self, multi_process_count: int):
+        """
+        Set number of process to be used in parallel for meshing the selected blade rows.
+
+        Parameters
+        ----------
+        multi_process_count:int
+            The number of processes to be used in parallel.
+        """
+        num_rows_to_process = len(self._blade_rows_to_mesh)
+        if num_rows_to_process == 0:
+            self._multi_process_count = 0
+            return
+        max_producers = min(cpu_count(), num_rows_to_process)
+        multi_process_count_clamped = min(max(0, multi_process_count), max_producers)
+        num_producers = min(num_rows_to_process, multi_process_count_clamped)
+        num_producers = num_producers if num_producers > 0 else max_producers
+        # print(f"{cpu_count()=}")
+        # print(f"{max_producers=}")
+        # print(f"{multi_process_count_clamped=}")
+        # print(f"{num_producers=}")
+        self._multi_process_count = num_producers
+
+    def set_blade_row_gsfs(self, assembly_gsf: float):
```

Comment:
I couldn't guess what this function was doing from its name as I had no idea what gsfs stood for. Can this function just be called "set_global_size_factor"?